src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,180 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newNode = new Node<>(null, value, null);
+            head = newNode;
+            tail = newNode;
+            size++;
+            return;
+        }
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail != null) {
+            tail.next = newNode;
+        } else {
+            head = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        verifyIndex(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0) {
+            head.prev = newNode;
+            newNode.next = head;
+            head = newNode;
+        } else {
+            Node<T> lookUpNode = findNodeByIndex(index);
+            lookUpNode.prev.next = newNode;
+            newNode.prev = lookUpNode.prev;
+            newNode.next = lookUpNode;
+            lookUpNode.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size) {
```

Comment:
Create separate method `indexValidation(index);` to avoid duplication in methods